js/woo.js

```diff
@@ -0,0 +1,984 @@
+            const maxQuote = this.safeNumber (market, 'quote_max');
+            const minBase = this.safeNumber (market, 'base_min');
+            const maxBase = this.safeNumber (market, 'base_max');
+            const priceScale = this.safeNumber (market, 'quote_tick');
+            const quantityScale = this.safeNumber (market, 'base_tick');
+            const minCost = this.safeNumber (market, 'min_notional');
+            result.push ({
+                'id': marketId,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'settle': undefined,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'settleId': undefined,
+                'type': marketTypeVal,
+                'spot': true,
+                'margin': false,
+                'swap': false,
+                'future': false,
+                'option': false,
+                'active': undefined,
+                'contract': false,
+                'linear': undefined,
+                'inverse': undefined,
+                // Fee is in %, so divide by 100
+                'taker': undefined,
+                'maker': undefined,
+                'contractSize': undefined,
+                'expiry': undefined,
+                'expiryDatetime': undefined,
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': {
+                    'amount': quantityScale,
+                    'price': priceScale,
+                },
+                'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    // I am dubious that something is not correct in my below prop-value assignments
+                    'amount': {
+                        'min': minBase,
+                        'max': maxBase,
+                    },
+                    'price': {
+                        'min': minQuote,
+                        'max': maxQuote,
+                    },
+                    'cost': {
+                        'min': minCost,
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        return result;
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchTrades() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchTrades', market, params);
```

Comment:
Do they have any other type of markets, other than `spot` ? If not, and it's always a `spot` market, then we should remove handleMarketTypeAndParams from here.

Reply:
They actually seem to have specific definitions for it. i.e. if you look their fetchMarkets response (their definition of symbols/pairs) they are even prefixed with `SPOT_BTC_USDT` and alike (They also have margin-related api responses). I think they are going to add other types too. for example, a while ago, binance also invested in them (and their marketing themselves as having 'deep pool'). So, I thought it would have been good in this case to keep them structured this way. But let me know